## Домашняя работа 2


# Задача 1.38

> Пятитомное собрание сочинений стоит на полке в случайном порядке.
> Какова вероятность того, что книги стоят слева направо в порядке нумерации томов?


Рассмотрим событие $A$ – книги стоят в правильном порядке.

Всего имеется 5 различных томов.

Так как используются все элементы и важен порядок их расположения, используем перестановки:

$P_n=n!$

Общее число вариантов расположения книг:

$n=5!$

Благоприятный вариант один:

$m=1$

In [1]:
import scipy.special as sc

n = sc.factorial(5)
m = 1
P_A = m / n

print("Вероятность события A:", round(P_A, 5))

Вероятность события A: 0.00833


# Задача 1.43

> Наудачу взятый телефонный номер состоит из 5 цифр.
> Найти вероятность того, что:
>
> а) цифры номера различны;
>
> б) все цифры одинаковые;
>
> в) все цифры нечётные.

Телефонный номер состоит из 5 цифр.

Первая цифра не может быть равна нулю.

Общее число возможных номеров:

$n=9\cdot10^4$

# a)
Рассмотрим событие $A$ – все цифры номера различны.

Первая цифра имеет 9 вариантов.

Остальные цифры: $9,8,7,6$ вариантов.

Количество благоприятных случаев:

$m_1=9\cdot9\cdot8\cdot7\cdot6$


In [4]:
import scipy.special as sc

n = 9 * 10**4
m1 = 9 * 9 * 8 * 7 * 6
P_A = m1 / n

print("Вероятность события A:", round(P_A, 4))

Вероятность события A: 0.3024


# б)
Рассмотрим событие $B$.

Одинаковыми могут быть цифры:

$1-9$

Всего благоприятных вариантов:

$m_2=9$

In [5]:
m2 = 9
P_B = m2 / n

print("Вероятность события B:", round(P_B, 4))

Вероятность события B: 0.0001


# в)
Нечётные цифры: $1,3,5,7,9$

Для каждой позиции существует 5 вариантов.
Количество благоприятных случаев:

$m_3=5^5$

In [6]:
m3 = 5**5
P_C = m3 / n

print("Вероятность события C:", round(P_C, 4))

Вероятность события C: 0.0347


# Задача 1.48

> При бросании трёх игральных костей найти вероятность:
>
> а) выпадения суммы 11 очков;
>
> б) выигрыша, если сумма очков больше 10.

In [7]:
from itertools import product

ls = list(product('123456', repeat=3))

print("Количество всех вариантов:", len(ls))

Количество всех вариантов: 216


# a)

Рассмотрим событие $A$ — при бросании трёх костей выпало 11 очков.

Общее количество исходов:

$n=6^3=216$

In [8]:
choice_11 = [
    el for el in ls 
    if sum(list(map(int, el))) == 11
]

print(f"Выпало 11 очков в {len(choice_11)} случаях.")

print(*choice_11, sep='\n')

Выпало 11 очков в 27 случаях.
('1', '4', '6')
('1', '5', '5')
('1', '6', '4')
('2', '3', '6')
('2', '4', '5')
('2', '5', '4')
('2', '6', '3')
('3', '2', '6')
('3', '3', '5')
('3', '4', '4')
('3', '5', '3')
('3', '6', '2')
('4', '1', '6')
('4', '2', '5')
('4', '3', '4')
('4', '4', '3')
('4', '5', '2')
('4', '6', '1')
('5', '1', '5')
('5', '2', '4')
('5', '3', '3')
('5', '4', '2')
('5', '5', '1')
('6', '1', '4')
('6', '2', '3')
('6', '3', '2')
('6', '4', '1')


Вероятность выпадения 11 очков

Количество благоприятных случаев:

$m=27$


In [10]:
P_11 = len(choice_11) / len(ls)

print("Вероятность выпадения 11 очков:", round(P_11, 4))

Вероятность выпадения 11 очков: 0.125


# б)

Рассмотрим событие $B$ — сумма очков больше 10.

Возможные суммы:

$11,12,13,14,15,16,17,18$

In [11]:
cnt = 0

for sm in range(11, 18+1):

    choice_sm = [
        el for el in ls 
        if sum(list(map(int, el))) == sm
    ]

    cnt += len(choice_sm)

print(f"Выпало больше 10 очков в {cnt} случаях.")

Выпало больше 10 очков в 108 случаях.


Количество благоприятных случаев:

$m=108$

Общее количество случаев:

$n=216$


In [12]:
P_win = cnt / len(ls)

print("Вероятность выигрыша:", round(P_win, 4))

Вероятность выигрыша: 0.5
